src/Conversion/ONNXToTOSA/Tensor/Reshape.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Reshape.cpp - Reshape Op ----------------------------===//
+//
+// Copyright (c) d-Matrix Inc. 2023
+//
+// =============================================================================
+//
+// This file lowers ONNX reshape operator to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToTOSA/DialectBuilder.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+class ONNXReshapeOpLoweringToTOSA : public OpConversionPattern<ONNXReshapeOp> {
+public:
+  using OpConversionPattern<ONNXReshapeOp>::OpConversionPattern;
+  LogicalResult matchAndRewrite(ONNXReshapeOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    TosaBuilder tosaBuilder(rewriter, op.getLoc());
+
+    RankedTensorType outputTy = op.getType().template cast<RankedTensorType>();
+    if (llvm::any_of(outputTy.getShape(), ShapedType::isDynamic))
+      return rewriter.notifyMatchFailure(op, "dynamic shapes not supported");
```

Comment:
This would now assert if the input is an unranked tensor. You could use `onnx_mlir::hasStaticShape ` to test both conditions at once and then use `notifyMatchFailure`